### PR TITLE
[FEATURE] emplace_exists

### PR DESCRIPTION
**Pull request by eseiler**
Needed for dynamic HIBF.
`emplace_exists` is around 5% slower than `emplace`.

### CODE REVIEW

File: include/hibf/interleaved_bloom_filter.hpp [hunk 1]
```diff
@@ -287,8 +293,6 @@ class interleaved_bloom_filter : private seqan::hibf::bit_vector
      * \param[in] new_bins_ The new number of bins.
      * \throws std::invalid_argument If passed number of bins is smaller than current number of bins.
      *
-     * \attention This function is only available for **uncompressed** Interleaved Bloom Filters.
-     * \attention The new number of bins must be greater or equal to the current number of bins.
```
Comment (smehringer) [inline include/hibf/interleaved_bloom_filter.hpp:291]: Is this removal intentional? Seems to be still valid.

File: src/interleaved_bloom_filter.cpp [hunk 2]
```diff
@@ -85,16 +85,39 @@ interleaved_bloom_filter::interleaved_bloom_filter(config & configuration) :
...
         idx += bin.value;
         assert(idx < size());
-        (*this)[idx] = 1;
+
+        // Constructing the reference twice for emplace_exists would impact performance.
+        // No difference for emplace.
+        seqan::hibf::bit_vector::reference bit_reference{(*this)[idx]};
+        if constexpr (check_exists)
+            exists &= bit_reference;
+        bit_reference = 1;
```
Comment (smehringer) [inline src/interleaved_bloom_filter.cpp:106]: Do you mean that

```
        if constexpr (check_exists)
            exists &= (*this)[idx];
        (*this)[idx] = 1;
```
Would construct the reference twice and is therefore slower?

**Review by smehringer — APPROVED** [review]
LGTM, just check if you really wanted to remove this one comment.

File: src/interleaved_bloom_filter.cpp (hunk 2 above)
Comment (eseiler) [reply src/interleaved_bloom_filter.cpp:106]: Yes, it would need to compute the bit position twice. Since there isn't much happening in `emplace`, doing it twice really affects performance.

```
emplace         50M/s
emplace_exists  47M/s
emplace_exists* 28M/s
```
`*` is with two references
